test/Chainweb/Test/Pact/Checkpointer.hs

```diff
@@ -369,6 +370,42 @@ checkpointerTest name initdata =
           runExec blockEnv09 Nothing "(m6.readTbl)" >>= \EvalResult{..} -> Right _erOutput @?= traverse toPactValue [tIntList [1,4]]
           _cpSave _cpeCheckpointer hash08
 
+          next "Don't create the same table twice! (part 1)"
+          hash09 <- BlockHash <$> merkleLogHash "0000000000000000000000000000009a"
+          blockEnv10 <- _cpRestore _cpeCheckpointer (Just (BlockHeight 9, hash08))
+
+          let h :: SomeException -> IO (Maybe String)
+              h = const (return Nothing)
+
+          let tKeyset = object ["test-keyset" .= object ["keys" .= ([] :: [Text]), "pred" .= String ">="]]
+          void $ runExec blockEnv10 (Just tKeyset) tablecode
+          expectedException1 <- (runExec blockEnv10 (Just tKeyset) tablecode >> return (Just "The table duplication somehow went through. Investigate this error.")) `catch` h
+          maybe (return ()) (flip assertBool False) expectedException1
```

Comment:
This should be just

`expectException $ runExec blockEnv10 (Just tKeyset) tablecode` rather than two lines of repeated boilerplate

On exception this will leak the db handle and will not discard the open transaction -- wrap in one of the `with-` functions